Sources/Testing/Events/Recorder/Event.AdvancedConsoleOutputRecorder.swift

```diff
@@ -31,7 +31,48 @@ extension Event {
       /// This is needed because ABI.EncodedEvent doesn't contain full test context.
       var testStorage: [String: ABI.EncodedTest<V>] = [:]
       
-      // Future storage for result data and other event information can be added here
+      /// Hierarchical test node structure for building the test tree.
+      var nodes: [String: _HierarchyNode] = [:]
+      var rootNodes: [String] = []
+      
+      /// Test results and timing information.
+      var testResults: [String: _TestResult] = [:]
+      var testTimings: [String: (start: ABI.EncodedInstant<V>?, end: ABI.EncodedInstant<V>?)] = [:]
+      var testIssues: [String: [ABI.EncodedIssue<V>]] = [:]
+      
+      /// Run timing information.
+      var runStartTime: ABI.EncodedInstant<V>?
+      var runEndTime: ABI.EncodedInstant<V>?
+      var isRunCompleted: Bool = false
```

Comment:
What is `isRunCompleted` used for? It only appears to be assigned, but never read